python/cuml/neighbors/nearest_neighbors.pyx

```diff
@@ -425,6 +314,139 @@ cdef class NearestNeighbors:
 
 
     def _kneighbors(self, X_ctype, N, k, I_ptr, D_ptr):
+
+
+        cdef uintptr_t inds = I_ptr
+        cdef uintptr_t dists = D_ptr
+        cdef uintptr_t x = X_ctype
+
+        self.k.search(<float*>x,
+                      <int> N,
+                      <long*>inds,
+                      <float*>dists,
+                      <int> k)
```

Comment:
There should also be a sync here before returning from the `kneighbors` call, no?

Reply:
The developer guide states the following:
```
cuML is thread safe so its functions can be called from multiple host threads if they use different handles.

The implementation of cuML is single threaded.
```

This seems to imply that we can expect a `cumlHandle` will not be issued across threads. In the case of the `kneighbors()` function, the OMP parallel call should create an implicit barrier, after which knn results from multiple GPUs are merged in host memory and results updated on device. 

Wouldn't this function already block cython execution until it is complete?